apps/openchallenges/mcp-server/src/main/java/org/sagebionetworks/openchallenges/mcp/server/config/McpServerConfigData.java

```diff
@@ -6,9 +6,27 @@
 
 @Data
 @Configuration
-@ConfigurationProperties(prefix = "openchallenges-mcp-server")
+@ConfigurationProperties
 public class McpServerConfigData {
 
-  private String welcomeMessage;
-  private String apiBaseUrl;
+  private Welcome welcome;
+  private Api api;
+
+  @Data
+  public static class Welcome {
+
+    private String message;
+  }
+
+  @Data
+  public static class Api {
+
+    private Base base;
+
+    @Data
+    public static class Base {
+
+      private String url;
+    }
+  }
```

Comment:
This is too much code to maintain. Instead, we should rely on `@Data` more as done previously. For now keep two private properties at the same level as there is no need to nest properties at this time.